cmd/wharf/run.go

```diff
@@ -85,13 +91,23 @@ https://iver-wharf.github.io/#/usage-wharfyml/`,
 			return err
 		}
 
-		ctx := context.Background()
+		ctx, cancel := context.WithCancel(context.Background())
```

Comment:
Suggest defining this in `root.go` as e.g `rootContext`, and reuse it in all of the subcommands (where applicable), such as in:

- `wharf aggregator serve`
- `wharf provisioner create`
- etc

Would be nice to also change `provisionerapi` and `watchdog` to support contexts as well, or create an issue on adding context support to them

Reply:
Changed to a context with cancel defined in `root.go`, and changed to use it instead of `context.Background` where I think it's correct.

Changed calling of the signal interception to `init` in `root.go`.